connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqConsumer.java

```diff
@@ -0,0 +1,117 @@
+import io.camunda.connector.impl.ConnectorInputException;
+import io.camunda.connector.rabbitmq.inbound.model.RabbitMqInboundResult;
+import io.camunda.connector.rabbitmq.inbound.model.RabbitMqInboundResult.RabbitMqInboundMessage;
+import io.camunda.connector.rabbitmq.supplier.GsonSupplier;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.apache.commons.text.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RabbitMqConsumer extends DefaultConsumer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMqConsumer.class);
+
+  private final InboundConnectorContext context;
+
+  public RabbitMqConsumer(Channel channel, InboundConnectorContext context) {
+    super(channel);
+    this.context = context;
+  }
+
+  @Override
+  public void handleDelivery(
+      String consumerTag, Envelope envelope, BasicProperties properties, byte[] body)
+      throws IOException {
+
+    LOGGER.debug("Received AMQP message with delivery tag {}", envelope.getDeliveryTag());
+    try {
+      RabbitMqInboundResult variables = prepareVariables(consumerTag, envelope, properties, body);
+      InboundConnectorResult<?> result = context.correlate(variables);
+
+      if (result != null && result.isActivated()) {
+        LOGGER.debug("ACK - inbound event correlated successfully: {}", result.getResponseData());
+        getChannel().basicAck(envelope.getDeliveryTag(), false);
+      } else {
+        LOGGER.debug("NACK (no requeue) - inbound event not correlated: {}", result.getErrorData());
+        getChannel().basicReject(envelope.getDeliveryTag(), false);
+      }
+
+    } catch (ConnectorInputException e) {
+      LOGGER.warn("NACK (no requeue) - failed to parse AMQP message body: {}", e.getMessage());
+      getChannel().basicReject(envelope.getDeliveryTag(), false);
+    } catch (Exception e) {
+      LOGGER.debug("NACK (requeue) - failed to correlate event", e);
+      getChannel().basicReject(envelope.getDeliveryTag(), true);
+    }
+  }
+
+  @Override
+  public void handleCancel(String consumerTag) {
+    LOGGER.info("Consumer cancelled: {}", consumerTag);
+    try {
+      context.cancel(null);
+    } catch (Exception e) {
+      LOGGER.error("Failed to cancel Connector execution: {}", e.getMessage());
+    }
+  }
+
+  @Override
+  public void handleShutdownSignal(String consumerTag, ShutdownSignalException sig) {
+    LOGGER.error("Consumer shutdown: {}", consumerTag, sig);
+    try {
+      context.cancel(sig);
+    } catch (Exception e) {
+      LOGGER.error("Failed to cancel Connector execution: {}", e.getMessage());
+    }
+  }
+
+  private RabbitMqInboundResult prepareVariables(
+      String consumerTag, Envelope envelope, BasicProperties properties, byte[] body) {
+
+    try {
+      String bodyAsString = new String(body, StandardCharsets.UTF_8);
+      JsonElement bodyAsJsonElement =
+          GsonSupplier.gson()
```

Comment:
Would be great to get rid of gson.

Reply:
Let's do it in a separate PR for all connectors